### PR TITLE
6 feature 6 chat room

### PR DESCRIPTION
**Pull request by 1StevenJiang1**

### CODE REVIEW

**Review by chazzybearr — CHANGES_REQUESTED** [review]
src/main/java/entities/ChatRoom.java:
---
> I think the method names AddMessage and GetMessages should be changed to follow java's naming convention (addMessage, getMessage).

src/main/java/entities/Message.java:
---
> In my opinion, when we store the author of a message, we should store the User object, rather than the name of the author. 

> Storing the name is ambiguous, as there could be many users with the same name, and once Message is instantiated, the only information we have on the author of the message is their name, which could lead to problems in the future. I'd be open to discussing this further.

src/test/java/entities/ChatRoomTest.java:
---
> I noticed that the program is creating a new instance of ChatRoom for every test. Although it may seem insignificant, I think putting this in a @ BeforeEach annotation would avoid repeated code.

> The method tearDown doesn't seem to do anything, so it can be removed.

src/test/java/entities/MessageTest.java:
---
> Same points as above.

src/test/java/usecases/ChatRoomInteractorTest.java:
---
> Same points as above.

> I think the test you currently have is a good start, but I think there are too few tests to guarantee that the method works as intended (String edge cases).

---
In general, I think importing wildcards (*) should be removed and replaced with explicit packages instead, as it is considered bad practice. It could cause naming conflicts and is ambiguous.

**Review by Dawei-He2002 — DISMISSED** [review]
All changes now addressed.

**Review by chazzybearr — DISMISSED** [review]
Looks good to me!

File: src/main/java/entities/ChatRoom.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class ChatRoom {
+    ArrayList<Message> messages;
```
Comment (kldtntg) [inline src/main/java/entities/ChatRoom.java:6]: I think the type here might better be List

**Review by kldtntg — DISMISSED** [review]

File: src/main/java/entities/ChatRoom.java (hunk 1 above)
Comment (nitvishn) [reply src/main/java/entities/ChatRoom.java:6]: Yes, I think that would be more appropriate.

**Review by nitvishn — DISMISSED** [review]
Other than changing the ArrayList instance variables to List, you're good. I'll approve this so that it can be merged, but you should make this change in the next commit.
